### DISCUSSION TITLE
How to attach label with a name given in the function argument to some content inside function?

### DISCUSSION THREAD

**Discussion by PgLoLo** [typst/typst#3458] [Q&A] [answered]
I am making custom Heading 1 styling that looks something like that:

```typst
#let Chapter(title, image_path, header_label) = {
  show heading: it => {
    assert(it.level == 1)

    [
      #text(size: 36pt)[Chapter *#counter(heading).display()*.]

      #text(size: 26pt)[#it.body]
    ]  
  }
  
  pagebreak(weak: true)
  
  align(alignment.horizon)[
    = #title. <#header_label>
  ]
  
image(image_path)
  
  pagebreak(weak: true)
}
```

This function takes heading title as an input, creates separate page with chapter number (counter of heading) and some additional information (image). Ideally I want to use this function like this:
```typst
#Chapter("Some Title" "some_image_path.jpg") <sec:chapter_2>
```
but as far, as I understand, currently there is no way to attach label to custom objects. Thus I pass an argument to `Chapter` function `header_label` that I am trying to attach as label to the standard heading syntax (see above in function definition):
```
 = #title. <#header_label>
```
But it does not work: I think, it creates label `"#header_label"` instead.
What is the correct way to achieve this?

**Comment by Julian-Wassmann** [comment, marked as answer]
You can call the `label` function and pass a string instead of using the dedicated syntax.

```typst
align(alignment.horizon)[
  #heading([#title.]) #label(header_label)
]
```

Pass a string as an argument to your Chapter function which is used for the label:
```typst
#Chapter("Some Title", "some_image_path.jpg", "sec:chapter_2")
```
